controllers/wildflyserver_controller_test.go

```diff
@@ -580,3 +583,711 @@ func TestWildFlyServerWithSecurityContext(t *testing.T) {
+	assert.Nil(stsStartupProbe.HTTPGet)
+}
+
+func TestWildFlyServerWithExecAndHttpProbes(t *testing.T) {
+	ctrl.SetLogger(zap.New(zap.UseDevMode(true)))
+	assert := testifyAssert.New(t)
+
+	livenessScriptName := "test-liveness-script.sh"
+	readinessScriptName := "test-readiness-script.sh"
+	os.Setenv("SERVER_LIVENESS_SCRIPT", livenessScriptName)
+	os.Setenv("SERVER_READINESS_SCRIPT", readinessScriptName)
+
+	livenessExec := &corev1.ExecAction{
+		Command: []string{
+			"/bin/sh",
+			"-c",
+			"exec liveness probe",
+		},
+	}
+
+	livenessHttp := &corev1.HTTPGetAction{
+		Path:   "/test/liveness",
+		Port:   intstr.FromString("0"),
+		Host:   "host-liveness",
+		Scheme: "HTTP",
+	}
+
+	livenessProbe := &wildflyv1alpha1.ProbeSpec{
+		ProbeHandler: wildflyv1alpha1.ProbeHandler{
+			Exec:    livenessExec,
+			HTTPGet: livenessHttp,
+		},
+	}
+
+	wildflyServer := &wildflyv1alpha1.WildFlyServer{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: wildflyv1alpha1.WildFlyServerSpec{
+			ApplicationImage: applicationImage,
+			Replicas:         replicas,
+			LivenessProbe:    livenessProbe,
+		},
+	}
+
+	// Objects to track in the fake client.
+	objs := []runtime.Object{
+		wildflyServer,
+	}
+
+	// Register operator types with the runtime scheme.
+	s := scheme.Scheme
+	s.AddKnownTypes(wildflyv1alpha1.GroupVersion, wildflyServer)
+
+	// Create a fake client to mock API calls.
+	cl := fake.NewClientBuilder().WithRuntimeObjects(objs...).Build()
+
+	// Create a WildFlyServerReconciler object with the scheme and fake client.
+	r := &WildFlyServerReconciler{
+		Client: cl,
+		Scheme: s,
+		Log:    ctrl.Log.WithName("test").WithName("WildFlyServer"),
+	}
+
+	// Mock request to simulate Reconcile() being called on an event for a
+	// watched resource .
+	req := reconcile.Request{
+		NamespacedName: types.NamespacedName{
+			Name:      name,
+			Namespace: namespace,
+		},
+	}
+
+	// Creating StatefulSet and loop waiting for reconcile to end
+	_, err := reconcileUntilDone(t, r, req, 4)
+
+	// Check if stateful set has been created and has the correct size.
+	statefulSet := &appsv1.StatefulSet{}
+	err = cl.Get(context.TODO(), req.NamespacedName, statefulSet)
+	require.NoError(t, err)
+	assert.Equal(replicas, *statefulSet.Spec.Replicas)
+	assert.Equal(applicationImage, statefulSet.Spec.Template.Spec.Containers[0].Image)
+
+	// check the Probes values. Exec should win against http
+	stsLivenessProbe := statefulSet.Spec.Template.Spec.Containers[0].LivenessProbe
+	assert.Equal(int32(60), stsLivenessProbe.InitialDelaySeconds)
+	assert.Equal(int32(0), stsLivenessProbe.TimeoutSeconds)
+	assert.Equal(int32(0), stsLivenessProbe.PeriodSeconds)
+	assert.Equal(int32(0), stsLivenessProbe.SuccessThreshold)
+	assert.Equal(int32(0), stsLivenessProbe.FailureThreshold)
+	assert.True(reflect.DeepEqual(livenessExec.Command, stsLivenessProbe.Exec.Command))
+	assert.Nil(stsLivenessProbe.HTTPGet)
+}
+
```

Comment:
Hello @tommaso-borgato I don't see it necessary since there is no interaction between those conditions in the source code. Each probe uses its own method, and the existence of one vs another does not affect to any logic.

We always invoke each of the methods See https://github.com/wildfly/wildfly-operator/blob/a4764d07a3afe3541d24cd3aa2367117f765bbcf/pkg/resources/statefulsets/statefulset.go#L110-L114  

Anyway, since tests are quick and the change is easy, we can add it, but it is unnecessary from my point of view.

Reply:
ok, no need to if they are totally independent code paths